src/App.tsx

```diff
@@ -1,61 +1,162 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import Box from '@mui/material/Box';
+import { Button, MenuItem } from '@mui/material';
+import TextField from '@mui/material/TextField';
+import FormHelperText from '@mui/material/FormHelperText';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function findUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: findUserById(todo.userId),
+  };
+});
+
+function findUserByName(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
 
 export const App = () => {
+  const defaultUserOption = 'Choose a user';
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [todos, setTodos] = useState(preparedTodos);
+  const [isTitleErrorOccured, setIsTitleErrorOccured] = useState(false);
+  const [
+    isSelectedUserErrorOccured,
+    setIsSelectedUserErrorOccured,
+  ] = useState(false);
+
+  const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (selectedUserId === 0 || title === '') {
+      if (title === '') {
+        setIsTitleErrorOccured(true);
+      }
+
+      if (selectedUserId === 0) {
+        setIsSelectedUserErrorOccured(true);
+      }
+
+      return;
+    }
+
+    const newUser = findUserByName(selectedUserId);
+
+    setTodos(currentTodo => {
+      const maxTodoId = Math.max(...currentTodo.map(todo => todo.id));
+
+      setTitle('');
+      setSelectedUserId(0);
+
+      return [
+        ...currentTodo,
+        {
+          id: maxTodoId + 1,
+          title,
+          completed: false,
+          userId: newUser ? newUser.id : null,
+          user: newUser,
+        },
+      ];
+    });
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsTitleErrorOccured(false);
```

Comment:
Why do we need to set this variable to false all the time ? Maybe, let's do it when `isTitleErrorOccured` is true ?